pkg/hierarchy/device.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2019 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hierarchy
+
+import (
+	"path/filepath"
+	"strings"
+)
+
+var sysFSDirectoryPath = "/sys/"
+
+// Device represents a blockdevice. This struct is used by hierachy pkg which is used to
+// get the necessary blockdevice hierarchy information
+type Device struct {
+	// Path of the blockdevice. eg: /dev/sda, /dev/dm-0
+	Path string
+}
+
+// DependentDevices represents all the dependent blockdevices of the given Device
+type DependentDevices struct {
+	// Parent is the parent device of the given blockdevice
+	Parent string
+
+	// Partitions are the partitions of this device if any
+	Partitions []string
+
+	// Holders is the slice of block-devices that are held by a given
+	// blockdevice
```

Comment:
can we add to the comment something like this:

> holders are represented in SysFS under the partition layer, so they are dependent on partitions

Reply:
@richardelling  Did not get this comment. What is meant by the partition layer here?

Reply:
@richardelling didn't get the meaning of this. `holders are represented under partition layer`.

Reply:
we’re living in the hierarchy that Linux thrust upon us. The word “slave” being used for dependents of the block device and “holders” used for dependents of the partitions (slices). This is very confusing for everyone, but since we’re exposing it to the world via yaml, we should be as clear as possible.